lib/packwerk/association_inspector.rb

```diff
@@ -45,5 +42,18 @@ def association?(node)
       method_name = Node.method_name(node)
       @associations.include?(method_name)
     end
+
+    def association_name_when_custom_class_name(arguments)
```

Comment:
This method returns a const node containing the class name, not the association name, so its name is a little misleading. How about something like `#custom_class_name_for_association` or just `#custom_class_name`?